frontend/src/pages/Worksheet.js

```diff
@@ -101,36 +101,15 @@ function Worksheet() {
     setCourseModal([false, '']);
   }, []);
 
-  // Check to see if this course is hidden
-  const isHidden = useCallback(
-    (season_code, crn) => {
-      for (let i = 0; i < hidden_courses.length; i++) {
-        if (
-          hidden_courses[i][0] === season_code &&
-          hidden_courses[i][1] === crn
-        )
-          return i;
-      }
-      return -1;
-    },
-    [hidden_courses]
-  );
-
   // Hide/Show this course
   const toggleCourse = useCallback(
-    (season_code, crn, hidden) => {
-      // Hide course
-      if (!hidden) {
-        setHiddenCourses([...hidden_courses, [season_code, crn]]);
-      }
-      // Show course
-      else {
-        let temp = [...hidden_courses];
-        temp.splice(isHidden(season_code, crn), 1);
-        setHiddenCourses(temp);
-      }
+    (crn) => {
+      let new_hidden_courses = Object.assign({}, hidden_courses);
+      if (hidden_courses[crn]) new_hidden_courses[crn] = false;
+      else new_hidden_courses[crn] = true;
+      setHiddenCourses(new_hidden_courses);
```

Comment:
you can actually pass a function into setHiddenCourses and remove the direct dependency on hidden_courses

Reply:
not too sure what u meant here, so I just pushed what my understanding was